src/domain/participants/participants.js

```diff
@@ -95,73 +99,81 @@ const putParticipantsErrorByTypeAndID = async () => {
   try {
     Logger.info('postParticipants::begin')
-    const type = req.params.Type
-    if (Object.values(Enums.type).includes(type)) {
-      const requesterParticipantModel = await participant.validateParticipant(req.headers['fspiop-source'])
+    const type = params.Type
+    if (Object.values(Enums.Accounts.PartyAccountTypes).includes(type)) {
+      const requesterParticipantModel = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE])
       if (requesterParticipantModel) {
-        const response = await oracle.oracleRequest(req)
+        const response = await oracle.oracleRequest(headers, method, params, undefined, payload)
         if (response && (response.data !== null || response.data !== undefined)) {
-          const payload = {
+          const responsePayload = {
             partyList: [
               {
                 partyIdType: type,
-                partyIdentifier: req.params.ID,
-                fspId: req.payload.fspId
+                partyIdentifier: params.ID,
+                fspId: payload.fspId
               }
             ],
-            currency: req.payload.currency
+            currency: payload.currency
           }
           const options = {
-            partyIdType: req.params.Type,
-            partyIdentifier: req.params.ID
+            partyIdType: params.Type,
+            partyIdentifier: params.ID
           }
-          await participant.sendRequest(req, req.headers['fspiop-source'], Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT, Enums.restMethods.PUT, payload, options)
+          if (!headers[Enums.Http.Headers.FSPIOP.DESTINATION] || headers[Enums.Http.Headers.FSPIOP.DESTINATION] === '') {
+            headers[Enums.Http.Headers.FSPIOP.DESTINATION] = payload.fspId
+          }
+          await participant.sendRequest(headers, headers[Enums.Http.Headers.FSPIOP.SOURCE], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_PUT, Enums.Http.RestMethods.PUT, responsePayload, options)
         } else {
-          // TODO: what happens when nothing is returned
-          await participant.sendErrorToParticipant(req, req.headers['fspiop-source'], Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT,
-            util.buildErrorObject(Errors.ErrorObject.ADD_PARTY_ERROR, [{key: type, value: req.params.ID}]))
+          await participant.sendErrorToParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT,
+            ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR).toApiErrorObject(), headers, params)
         }
       } else {
         Logger.error('Requester FSP not found')
-        // TODO: handle issue where requester fsp not found. Pass to error handling framework
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()
```

Comment:
Same issue applies here

`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found')`